service/raft/raft_group0_client.cc

```diff
@@ -502,4 +502,100 @@ template group0_command raft_group0_client::prepare_command(write_mutations chan
 template group0_command raft_group0_client::prepare_command(broadcast_table_query change, std::string_view description);
 template group0_command raft_group0_client::prepare_command(write_mutations change, std::string_view description);
 
+api::timestamp_type mutations_collector::write_timestamp() const {
+    if (!_guard) {
+        on_internal_error(logger, "mutations_collector: write_timestamp without guard taken");
+    }
+    return _guard->write_timestamp();
+}
+
+void mutations_collector::add_mutation(mutation m) {
+    _muts.push_back(std::move(m));
+}
+
+void mutations_collector::add_mutations(std::vector<mutation> ms) {
+    _muts.insert(_muts.end(),
+            std::make_move_iterator(ms.begin()),
+            std::make_move_iterator(ms.end()));
+}
+
+void mutations_collector::add_generator(generator_func f) {
+    _generators.push_back(std::move(f));
+}
+
+static future<> add_write_mutations_entry(
+        ::service::raft_group0_client& group0_client,
+        std::string_view description,
+        std::vector<canonical_mutation> muts,
+        ::service::group0_guard group0_guard,
+        seastar::abort_source* as,
+        std::optional<::service::raft_timeout> timeout) {
+    logger.trace("add_write_mutations_entry: {} mutations with description {}",
+            muts.size(), description);
+    auto group0_cmd = group0_client.prepare_command(
+        ::service::write_mutations{
+            .mutations{std::move(muts)},
+        },
```

Comment:
not sure what you're suggesting here?

Reply:
Nothing in particular yet. However, if https://github.com/scylladb/scylladb/pull/18758 is merged before your changes, then you will have to use the `service_levels_change` command when applying changes to service levels, not `write_mutations`. I just wanted to point that out because the current state of the PR hardcodes the use of `write_mutations`.

Reply:
let's hurry up then :D

Reply:
but really we need to think it through it's not composable you if you use your specific command, do you need single transaction semantics for service level controller configuration change? Maybe you can emit your command without mutations so you achieve single push of config change but I still will be able to pack my mutations together in a separate command? cc @Jadw1

Reply:
@piodul are we good with resolving this conversation or you have further comments?